File: Programming Basics with Python/05 While loop/More exercises/01_Dishwasher.py
bottles = int(input())
detergent = bottles * 750
wash_counter = 0
dish_counter = 0
pot_counter = 0
while detergent >= 0:
    wash = input()
    if wash == "End":
        break
    wash_counter += 1
    if wash_counter == 3:
        detergent -= 15 * (int(wash))
        wash_counter = 0
        pot_counter += int(wash)
    else:
        detergent -= 5 * (int(wash))
        dish_counter += int(wash)
if detergent < 0:
    print(f"Not enough detergent, {abs(detergent)} ml. more necessary!")
else:
    print(
        f"Detergent was enough!\n{dish_counter} dishes and {pot_counter} pots were washed.\nLeftover detergent {detergent} ml.")

File: JS Basics/05 While-Loop - More Exercises/01. Dishwasher.js
function dishwasher(input) {
    let detergentBottles = Number(input[0]);
    let detergent = detergentBottles * 750;
    let index = 1;
    let counter = 0;
    let plates = 0;
    let pots = 0;
    while (true) {
        counter++;
        let washing = input[index];
        index++;
        if (washing == "End" || detergent < 0) {
            break;
        }
        if (counter % 3 == 0) {
            detergent -= Number(washing) * 15;
            pots += Number(washing)
        }
        else {
            detergent -= Number(washing) * 5;
            plates += Number(washing)
        }
    }
    if (detergent >= 0) {
        console.log("Detergent was enough!");
        console.log(`${plates} dishes and ${pots} pots were washed.`);
        console.log(`Leftover detergent ${detergent} ml.`);
    }
    else {
        console.log(`Not enough detergent, ${Math.abs(detergent)} ml. more necessary!`);
    }
}
dishwasher([2, 53, 65, 55, "End"])